File: ROS/day1/hw2/hw2_package/my_first_ros_rclpy_pkg/my_first_ros_rclpy_pkg/pub.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String
from std_msgs.msg import Int32
from std_msgs.msg import Float64


class HelloworldPublisher(Node):
    def __init__(self):
        super().__init__('helloworld_publisher')
        self.helloworld_publisher1 = self.create_publisher(String, 'topicname', 10)
        self.helloworld_publisher2 = self.create_publisher(Int32, 'topicname_int', 30)
        self.helloworld_publisher3 = self.create_publisher(Float64, 'topicname_float', 30)
        self.timer = self.create_timer(1, self.publish_helloworld_msg)
        self.count=0

    def publish_helloworld_msg(self):
        msg1 = String()
        msg2 = Int32()
        msg3 = Float64()
        msg1.data = 'Hello World: {0}'.format(self.count)
        msg2.data = self.count
        msg3.data = self.count/3
        self.helloworld_publisher1.publish(msg1)
        self.helloworld_publisher2.publish(msg2)
        self.helloworld_publisher3.publish(msg3)
        self.get_logger().info('Published message: {0}'.format(msg1.data))
        self.get_logger().info('Published message: {0}'.format(msg2.data))
        self.get_logger().info('Published message: {0}'.format(msg3.data))
        self.count+=1


def main(args=None):
    rclpy.init(args=args)
    node = HelloworldPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ROS/day2/hw1/custom_interfaces/src/pub.cpp

#include <chrono>
#include <functional>
#include <memory>
#include <string>


#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "std_msgs/msg/int64.hpp"
#include "std_msgs/msg/float64.hpp"

using namespace std::chrono_literals;
using namespace std;

class Pub : public rclcpp::Node
{
    public:
       Pub() : Node("pub")
       {
           publisher = this->create_publisher<std_msgs::msg::Int64>("topicname", 10);
           timer = this->create_wall_timer(10s, std::bind(&Pub::cmd_user_inp,this));
       }
       int count=0;

    private:
        rclcpp::TimerBase::SharedPtr timer;
       void cmd_user_inp(){
           auto msg = std_msgs::msg::Int64();
           cout<<"아무거나 정수나 입력하세요(10초 후 초기화) :  ";
           cin>>msg.data;
           RCLCPP_INFO(this->get_logger(), "Published message: '%ld'", msg.data);
           publisher->publish(msg);

       }
       rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr publisher;
};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv); 
    auto node = std::make_shared<Pub>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

